include/GPM/Shape3D/AABB.hpp

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ * found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "../Tools.hpp"
+#include "Volume.hpp"
+#include "../Vector3.hpp"
+
+namespace GPM
+{
+class AABB : public Volume
+{
+protected:
+    Vec3  m_center;
+    float m_iI, m_iJ, m_iK;
+
+public:
+
+    AABB()                  = default;
+    AABB(const AABB& other) = default;
+    AABB(AABB&& other)      = default;
+    virtual ~AABB()         = default;
+    AABB& operator=(AABB const& other) = default;
+    AABB& operator=(AABB&& other) = default;
+
+    AABB(const Vec3& center, float iI, float iJ, float iK) : Volume{}, m_center{center}, m_iI{iI}, m_iJ{iJ}, m_iK{iK}
+    {
+    }
+
+    bool isInside(Vec3 pt) const noexcept
+    {
+        Vec3        localPt = pt - m_center;
+        const float espilon =
+            0.0001f; /*The standard float espsilone his enought in this case. The float value has chang to much*/
+
+        return isBetween(localPt.x, -m_iI - espilon, m_iI + espilon) &&
+               isBetween(localPt.y, -m_iJ - espilon, m_iJ + espilon) &&
+               isBetween(localPt.z, -m_iK - espilon, m_iK + espilon);
+    }
```

Comment:
Consider renaming this method. If I were reading:
```C++
AABB aabb;
Vec3 pt;
bool b = aabb.isInside(pt);
```
I would interpret `b` as: "is the AABB inside the point?"

I'm not sure what the `epsilon` is for here. The interval of comparison will always be large enough to be superior to `epsilon`, since an AABB with `m_iI = 0.f`, `m_iJ = 0.f` or `m_iK = 0.f` is useless

```suggestion
    bool isInside(const Vec3& pt) const noexcept
    {
        const Vec3 localPt{pt - m_center};
        
        return isBetween(localPt.x, -m_iI, m_iI) &&
               isBetween(localPt.y, -m_iJ, m_iJ) &&
               isBetween(localPt.z, -m_iK, m_iK);
    }
```